x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/delete_rules_route.ts

```diff
@@ -43,9 +46,17 @@ export const createDeleteRulesRoute: Hapi.ServerRoute = {
         id,
         ruleId,
       });
-
+      const ruleStatuses = await savedObjectsClient.find({
+        type: ruleStatusSavedObjectType,
+        perPage: 10,
+        search: `"${id}"`,
+        searchFields: ['alertId'],
+      });
+      ruleStatuses.saved_objects.forEach(async obj =>
+        savedObjectsClient.delete(ruleStatusSavedObjectType, obj.id)
+      );
       if (rule != null) {
-        return transformOrError(rule);
+        return transformOrError(rule, ruleStatuses.saved_objects[0]);
```

Comment:
You want to move the above code because rule _could_ be `null` right in this check like so:

```ts
      const rule = await readRules({
        alertsClient,
        id,
        ruleId,
      });
      if (rule != null) {
        const ruleStatuses = await savedObjectsClient.find({
          type: ruleStatusSavedObjectType,
          perPage: 10,
          search: `"${rule.id}"`,
          searchFields: ['alertId'],
        });
        ruleStatuses.saved_objects.sort((a, b) => {
          const dateA = new Date(a.attributes.statusDate);
          const dateB = new Date(b.attributes.statusDate);
          if (dateA < dateB) {
            return 1;
          } else if (dateA === dateB) {
            return 0;
          }
          return -1;
        });
        return transformOrError(rule, ruleStatuses); // update this to run with an array of rule statuses
      } else {
        return getIdError({ id, ruleId });
      }
    } catch (err) {
      return transformError(err);
    }
```